deployments/kovan/usdc/deploy.ts

```diff
@@ -1,251 +1,176 @@
+  await deploymentManager.idempotent(
+    async () => !sameAddress(await governor.timelock(), timelock.address),
+    async () => {
+      debug(`Initializing GovSimple`);
+      await wait(governor.initialize(timelock.address, [signer.address]));
+    }
   );
 
-  let link = await deploymentManager.clone(
-    cloneAddr.link,
-    [],
+  // Deploy UNI first because it is the flakiest (has a dependency on block timestamp)
+  // TODO: currently this retries with the same timestamp. we should update the timestamp on retries
+  const UNI = await deploymentManager.clone(
+    'UNI',
+    cloneAddr.uni,
+    [signer.address, signer.address, (await getBlock(null, ethers)).timestamp + 60],
     cloneNetwork
   );
 
-  // Contracts referenced in `configuration.json`.
-  let contracts = new Map<string, Contract>([
-    ['USDC', usdc],
-    ['WBTC', wbtc],
-    ['WETH', weth],
-    ['COMP', comp],
-    ['UNI', uni],
-    ['LINK', link],
-  ]);
+  const usdcProxyAdmin = await deploymentManager.deploy('USDC:admin', 'vendor/proxy/transparent/ProxyAdmin.sol', []);
+  const usdcImpl = await deploymentManager.clone('USDC:implementation', cloneAddr.usdcImpl, [], cloneNetwork);
+  const usdcProxy = await deploymentManager.clone('USDC', cloneAddr.usdcProxy, [usdcImpl.address], cloneNetwork);
+  const usdcProxyAdminSlot = '0x10d6a54a4754c8869d6886b5f5d7fbfa5b4522237ea5c60d11bc4e7a1ff9390b';
+  const USDC = usdcImpl.attach(usdcProxy.address);
+
+  await deploymentManager.idempotent(
+    async () => !sameAddress(await ethers.provider.getStorageAt(usdcProxy.address, usdcProxyAdminSlot), usdcProxyAdmin.address),
+    async () => {
+      debug(`Changing admin of USDC proxy to ${usdcProxyAdmin.address}`);
+      await wait(usdcProxy.connect(signer).changeAdmin(usdcProxyAdmin.address));
+
+      debug(`Initializing USDC`);
+      await wait(USDC.connect(signer).initialize(
+        'USD Coin',     // name
+        'USDC',         // symbol
+        'USD',          // currency
+        6,              // decimals
+        signer.address, // Master Minter
+        signer.address, // Pauser
+        signer.address, // Blacklister
+        signer.address  // Owner
+      ));
+    }
+  );
+
+  const WBTC = await deploymentManager.clone('WBTC', cloneAddr.wbtc, [], cloneNetwork);
+  const WETH = await deploymentManager.clone('WETH', cloneAddr.weth, [], cloneNetwork);
+  const COMP = await deploymentManager.clone('COMP', cloneAddr.comp, [signer.address], cloneNetwork);
+  const LINK = await deploymentManager.clone('LINK', cloneAddr.link, [], cloneNetwork);
 
   // Deploy all Comet-related contracts
-  let { cometProxy, configuratorProxy, timelock, rewards } = await deployComet(
-    deploymentManager,
-    { all: true },
-    {},
-    contracts
-  );
+  const deployed = await deployComet(deploymentManager, deploySpec);
+  const { comet } = deployed;
 
   // Deploy Bulker
-  const bulker = await deploymentManager.deploy<Bulker, Bulker__factory, [string, string, string]>(
```

Comment:
We should wrap this around a retry as well right?

Reply:
Deploy retries